### PR TITLE
Add Matrix unittests

### PR DESCRIPTION
**Pull request by dktcoding**
As promised, here are the Unit Tests for the Matrix class, for the other classes I mostly want to write integration tests.
The output for `ant test` was:
```
-do-test-run:
    [junit] Testsuite: libai.common.MatrixTest
    [junit] Tests run: 52, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.029 sec
```

### CODE REVIEW

File: nbproject/project.properties [hunk 1]
```diff
@@ -30,24 +30,23 @@ dist.jar=${dist.dir}/libai.jar
 dist.javadoc.dir=${dist.dir}/javadoc
 endorsed.classpath=
 excludes=
-file.reference.mysql-connector-5.0.5.jar=../../libs/mysql/mysql-connector-5.0.5.jar
 includes=**
 jar.compress=false
 javac.classpath=\
-    ${libs.swing-layout.classpath}:\
-    ${file.reference.mysql-connector-5.0.5.jar}
```
Comment (kronenthaler) [inline nbproject/project.properties]: This dependency is required for decision trees at the moment. In the decision tree package there is an interesting structure called DataSet that i want to use for the training of neural networks as it allows to do creation of training sets using cross references percentages.

File: test/libai/common/MatrixTest.java [hunk 2]
```diff
@@ -0,0 +1,788 @@
...
+            test[i] = 1;
+            assertArrayEquals(test, m.getRow(i), DELTA);
+        }
+    }
+
+    @Test
+    public void testMatrixRandom() {
+        Matrix a = Matrix.random(50, 100, true);
+        Matrix b = Matrix.random(50, 100, false);
+        assertNotEquals(a, b); //<- yes... this test could actually fail...
```
Comment (kronenthaler) [inline test/libai/common/MatrixTest.java]: insert a small delay, Thread.sleep(1), between creations should be enough to offset the creations.

File: nbproject/project.properties (hunk 1 above)
Comment (dktcoding) [inline nbproject/project.properties]: Oh sorry about that, I clicked on "Resolve dependencies" and didn't realize that the connector jar was gone.

I've been looking at the `DataSet` class and it uses `javax.sql.Connection` which is driver agnostic in which case the MySQL Connector/J should be imported by whoever is using `libai` and not `libai` itself.

At any rate I will fix it now because it was an unintended change.

File: test/libai/common/MatrixTest.java [hunk 3]
```diff
@@ -0,0 +1,789 @@
...
+        for (int i = 0; i < m.getRows(); i++) {
+            double[] test = new double[15];
+            test[i] = 1;
+            assertArrayEquals(test, m.getRow(i), DELTA);
+        }
+    }
+
+    @Test
+    public void testMatrixRandom() {
+        Matrix a = Matrix.random(50, 100, true);
```
Comment (kronenthaler) [inline test/libai/common/MatrixTest.java]: Another alternative is to use a seed as parameter in this method, this also will make the code more testable for other parts.

Comment (dktcoding) [inline test/libai/common/MatrixTest.java]: Agreed, adding a `Random` object as a parameter would simplify things, actually was a part of the comment of #13. 

There are a couple of ways of actually implementing it, I'll create a small commit as a starting point.

Comment (kronenthaler) [inline test/libai/common/MatrixTest.java]: I will merge this PR as it's fine, we introduce those changes separately to keep proper track of them
